Classes/Utility/FrontendUtility.php

```diff
@@ -185,21 +185,21 @@ public static function isCacheable($conf = null)
     /**
      * Return current URL
      *
-     * @return null|string
+     * @return string
      */
     public static function getCurrentUrl()
     {
-        $ret = null;
-
-        $TSFE = self::getTsfe();
+        //former $TSFE->anchorPrefix got deprecated in TYPO3 7.x
+        $tsfeAnchorPrefix = substr(
+            Typo3GeneralUtility::getIndpEnv('TYPO3_REQUEST_URL'),
```

Comment:
But for `'TYPO3_SITE_SCRIPT'`, TYPO3 does a

    case 'TYPO3_SITE_SCRIPT':
        $retVal = substr(self::getIndpEnv('TYPO3_REQUEST_URL'), strlen(self::getIndpEnv('TYPO3_SITE_URL')));

... which is pretty much the same, just shorter.